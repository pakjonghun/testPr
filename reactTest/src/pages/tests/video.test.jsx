import {
  screen,
  render,
  waitForElementToBeRemoved,
} from '@testing-library/react';
import { Route } from 'react-router-dom';
import { withAllContext, withRoute } from '../../tests/utils';
import Videos from '../Videos';
import { videos } from '../../tests/videos';

const jestFn = jest.fn();

const youtube = {
  search: jestFn,
};

describe('video', () => {
  afterEach(() => jestFn.mockReset());
  it('render correctly', async () => {
    jestFn.mockImplementation(() => videos);

    const { asFragment } = render(
      withAllContext(withRoute(<Route path="/" element={<Videos />} />), {
        youtube,
      })
    );
    await waitForElementToBeRemoved(() => screen.queryByText('Loading...'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('error', async () => {
    jestFn.mockImplementation(() => {
      throw new Error('error');
    });

    render(
      withAllContext(withRoute(<Route path="/" element={<Videos />} />), {
        youtube,
      })
    );
    const err = 'Something is wrong ðŸ˜–';

    await screen.findByText(err);
  });

  it('have video', async () => {
    const keyword = 'keyword';

    jestFn.mockImplementation(() => videos);

    render(
      withAllContext(
        withRoute(
          <>
            <Route path="/:keyword" element={<Videos />} />
          </>,
          [`/${keyword}`]
        ),
        { youtube }
      )
    );

    await waitForElementToBeRemoved(() => screen.queryByText('Loading...'));
    expect(jestFn).toBeCalledWith(keyword);

    videos.forEach((video) => {
      const url = video.snippet.thumbnails.medium.url;
      expect(screen.getByRole('img')).toHaveAttribute('src', url);
    });
  });
});
