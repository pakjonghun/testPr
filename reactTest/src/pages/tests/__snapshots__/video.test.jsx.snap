// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`video render correctly 1`] = `
<DocumentFragment>
  <ul
    class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 2xl:grid-cols-5 gap-2 gap-y-4"
  >
    <li
      class=""
    >
      <img
        alt="Survive 100 Days In Circle, Win $500,000"
        class="w-full"
        src="https://i.ytimg.com/vi/gHzuabZUd6c/mqdefault.jpg"
      />
      <div>
        <p
          class="font-semibold my-2 line-clamp-2"
        >
          Survive 100 Days In Circle, Win $500,000
        </p>
        <p
          class="text-sm opacity-80"
        >
          aa
        </p>
        <p
          class="text-sm opacity-80"
        >
          2 months ago
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
